### ISSUE TITLE
espanso stops expanding when a shell command from another expansion is still running

### ISSUE THREAD

**Issue by isorensen** [federico-terzi/espanso#728]
# Summary

I'm trying to create a shortcut for editing my config files, such as xmonad, xmobar, conky, etc.
I create a shell command to launch alacrity with vim, and as long the file is still opened, other calls for text expansions do not work.

## Steps to reproduce

1. Create a shell trigger that calls a terminal with vim
```yml
  - trigger: "!!xmobar"
    replace: "{{output}}"
    vars:
      - name: output
        type: shell
        params:
          cmd: "alacritty -e vim $HOME/.config/xmobar/xmobarrc.hs"
```

2. Save yml file, and restart espanso with `espanso restart`

3. Call the expansion `!!xmobar`

4. Try to expand another trigger, such as `:espanso`

## Expected behavior

`:espanso` would expand to "Hi There!"

## Current behavior

`:espanso` does not expand

## Relevant information

After quitting vim loaded with xmonad.hs, espanso works once again normally.

## What I think could be happening

Espanso daemon gets "stuck" with just one expansion, and waits until it completely finishes to be able to call another expansion.

## Doubts

Is there another way I should triggering this command, that would actually make espanso keep working? Or this issue is actually something that needs to be targeted?

**Comment by federico-terzi** [comment]
Hey @isorensen,

Thank you for the feedback! This is the expected default behavior, as shell scripts are run synchronously (aka, we want the output of the shell script to expand the snippet).

I'm wondering if you could make this work by adding the `&` at the end, like so (not tested):

```yaml
  - trigger: "!!xmobar"
    replace: "{{output}}"
    vars:
      - name: output
        type: shell
        params:
          cmd: "alacritty -e vim $HOME/.config/xmobar/xmobarrc.hs &"
```

which should start the command in the background. Let me know if that helps

Cheers :)

**Comment by isorensen** [comment]
Hello @federico-terzi , thank you for the response. Adding `&` didn´t work. I´d actually have tried it already. I understood it's the expected default behavior. I'll leave it to you to decide weather this is a feature to be addressed, or just close the issue, ok?  
On my side, I´ll still keep using this command, since, usually, I can do my edits in these config files without actually needing other expansion snippets. I do think it would be nice! :-)

Best regards, and congratulations for the great tool.

**Comment by federico-terzi** [comment]
@isorensen I see, thank you! I think this use-case will be solved once #556 will be implemented, as there I can definitely see an option to spawn processes in a non-synchronous way. For now, I'll close the issue, let's continue the discussion there :)
